pkg/pr/updates.go

```diff
@@ -84,6 +91,70 @@ func processRegexReplacements(replacements []RegexReplacement, ctx map[string]in
 	return nil
 }
 
+func processYamlOverlays(overlays []YamlOverlay, ctx map[string]interface{}) error {
+	if len(overlays) == 0 {
+		return nil
+	}
+
+	for _, overlay := range overlays {
+		var err error
+		var overlayYaml = []byte(overlay.Yaml)
+
+		if overlay.Templated {
+			overlayYaml, err = templateReplacement([]byte(overlay.Yaml), ctx)
+			if err != nil {
+				return err
+			}
+		}
+
+		mergeFunc := func(data []byte) ([]byte, error) {
+			return mergeYaml(data, overlayYaml, overlay.ListMerge)
+		}
+
+		if err = replaceInPlace(overlay.File, mergeFunc); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func mergeYaml(base, overlay []byte, merge ListMerge) ([]byte, error) {
+	baseMap := make(map[string]interface{})
+	overlayMap := make(map[string]interface{})
+
+	if err := yaml.Unmarshal(base, &baseMap); err != nil {
+		return nil, err
+	}
+
+	if err := yaml.Unmarshal(overlay, &overlayMap); err != nil {
+		return nil, err
+	}
+
+	options := []func(*mergo.Config){mergo.WithOverride, mergo.WithSliceDeepCopy}
+	if merge == ListMergeAppend {
```

Comment:
We should make this case insensitive (just `strings.ToUpper(merge)` it).  There's an annoying thing where internally in the elixir code this is downcased but it is presented via gql as uppercase that could cause bugs.

Reply:
Oh wait i misread, it is being upcased

Reply:
It should be already handled by https://github.com/pluralsh/plural-cli/pull/573/files#diff-93646044a7d487363a18f90d1c82940c54ca0961fdc2a290a4799884bfcc40eaR37-R51. When it will be converted from the CRD to this internal type, it will do case insensitive comparison.